### PR TITLE
Manually append a trailing slash at the end of a path if needed

### LINKED ISSUE #60: cli.BuildURL: Allow for trailing slash at the end of the path
Some endpoints (e.g. https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#get-matrix-client-r0-pushrules) require a trailing slash at the end of the path, but `cli.BuildURL` seems to strip it from the built URL. Possibly tricky as it might come from the `url` package.

### PR DESCRIPTION
**Pull request by babolivier**
Fixes #60

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -68,6 +69,10 @@ func (cli *Client) BuildBaseURL(urlPath ...string) string {
 	parts := []string{hsURL.Path}
 	parts = append(parts, urlPath...)
 	hsURL.Path = path.Join(parts...)
+	// Manually add the trailing slash back to the end of the path if it's explicitely needed
```
Comment (richvdh) [inline client.go]:
```suggestion
	// Manually add the trailing slash back to the end of the path if it's explicitly needed
```

**Comment by babolivier** [comment]
> Out of interest, which paths are affected?

I don't have a specific list, but this impacted any path that ends with a trailing slash in the spec (as it was systematically removing it).

**Comment by richvdh** [comment]
do you have any examples? Basically I am surprised that we have any APIs that require trailing slashes.

**Comment by babolivier** [comment]
Gah, missed the notif for your message. An example is https://matrix.org/docs/spec/client_server/r0.4.0.html#get-matrix-client-r0-pushrules

**Comment by richvdh** [comment]
> Gah, missed the notif for your message. An example is https://matrix.org/docs/spec/client_server/r0.4.0.html#get-matrix-client-r0-pushrules

thanks, raised as https://github.com/matrix-org/matrix-doc/issues/1944.

**Pull request merged**